Message送信機能の実装

avaklee opened the pull request:
# WHAT
メッセージ送信機能の実装
# WHY
必要なので

avaklee committed:
Message送信機能の実装

# WHAT
メッセージ送信機能の実装
# WHY
必要なので

seoky0 commented on app/controllers/messages_controller.rb:
```diff
@@ -1,6 +1,17 @@
+  def create
+    @message = Message.new(message_params)
+    if @message.save
+      redirect_to controller: 'groups', action: 'index'
```
リダイレクト先はプレフィックスで指定しましょう

seoky0 commented on app/controllers/messages_controller.rb:
```diff
@@ -1,6 +1,17 @@
+    if @message.save
+      redirect_to controller: 'groups', action: 'index'
+    else
+      flash[:alert] = "メッセージを入力してください。"
```
どの画面を描画するか決めずに大丈夫でしょうか？

seoky0 commented on app/models/message.rb:
```diff
@@ -1,4 +1,6 @@
 class Message < ApplicationRecord
 belongs_to :user
 belongs_to :group
+validates :body, presence: true
```
バリデーションは、文章も画像もない場合に投稿できないようにしましょう

avaklee committed:
message送信機能修正

# WHAT
バリデーションの設定とレンダーの追加
# WHY
Image or bodyどちらかがあればokになるようにするため、レンダーは前回のミス

kinshotomoya commented on app/controllers/messages_controller.rb:
```diff
@@ -3,9 +3,10 @@ class MessagesController < ApplicationController
+      redirect_to root_path
     else
       flash[:alert] = "メッセージを入力してください。"
+      render　:create
```
createですか？
メッセージを新しく作る画面に変移です

kinshotomoya commented on app/models/message.rb:
```diff
@@ -1,6 +1,8 @@
 belongs_to :user
 belongs_to :group
-validates :body, presence: true
+validates :body, :image, presence: true
```
この記述いらないですね。
これだと、body imageどっちもないとバリデーションに引っかかってしまいますね！
下記のコードで十分です！

avaklee replied:
すみません。反映しなかったみたいで。
validates :body, presence: true, if: -> { image.blank? }
validates :image, presence: true, if: -> { body.blank? }

こう書きました。

kinshotomoya reviewed (commented):
コメントしました！
下記、修正してくださイ！！！

avaklee committed:
Message送信機能の実装

# What
Review頂いた、Render先の修正、validationの記載（前回saveし忘れてた部分）、gitignoreの追加

#Why
間違いの修正・gitignoreは必要なので

seoky0 commented:
コミットメッセージには、何をしたのかを記述してください。
〜の実装ではなく、実装の中で何をしたか（ビューの変更、コントローラの修正、メソッドの定義など）
また、修正もどのように修正したのか記述いただくようにお願いいたします。
